optax/_src/loss.py

```diff
@@ -519,3 +519,58 @@ def hinge_loss(predictor_outputs: chex.Array,
     Binary Hinge Loss.
   """
   return jnp.maximum(0, 1 - predictor_outputs * targets)
+
+
+def poly_loss_cross_entropy(logits: chex.Array,
+                            labels: chex.Array,
+                            epsilon: float = 2.0,
+                            alpha: float = 0.0) -> chex.Array:
+  r"""Computes PolyLoss with alpha-label smoothing between logits and labels.
+
+  The PolyLoss is a loss function that decomposes commonly
+  used classification loss functions into a series of weighted
+  polynomial bases. It is inspired by the Taylor expansion of
+  cross-entropy loss and focal loss in the bases of :math:`(1 − P_t)^j`.
+
+  .. math::
+    L_{Poly} = \sum_1^\infty \alpha_j \cdot (1 - P_t)^j \\
+    L_{Poly-N} = (\epsilon_1 + 1) \cdot (1 - P_t) + \ldots + \\
+    (\epsilon_N + \frac{1}{N}) \cdot (1 - P_t)^N +
+    \frac{1}{N + 1} \cdot (1 - P_t)^{N + 1} + \ldots = \\
+    - \log(P_t) + \sum_{j = 1}^N \epsilon_j \cdot (1 - P_t)^j
+
+  This function provides a simplified version of the :math:`L_{Poly-N}`
+  with only the coefficient of the first polynomial term being changed.
+
+    References:
+      [Zhaoqi Leng et al, 2022](https://arxiv.org/pdf/2204.12511.pdf)
+
+    Args:
+      logits: Unnormalized log probabilities, with shape `[..., num_classes]`.
+      labels: Valid probability distributions (non-negative, sum to 1), e.g a
+        one hot encoding specifying the correct class for each input;
+        must have a shape broadcastable to `[..., num_classes]``
+      epsilon: The coefficient of the first polynomial term (default = 2.0).
+        According to the paper, the following values are recommended:
+        - For the ImageNet 2d image classification, epsilon = 2.0
+        - For the 2d Instance Segmentation and object detection, epsilon = -1.0
+        - It is also recommended to adjust this value
+          based on the task and dataset at hand. For example, one can use
+          simple grid search to achieve it.
+      alpha: The smoothing factor, the greedy category with be assigned
```

Comment:
My bad, I copypasted the description from [here](https://github.com/deepmind/optax/blob/master/optax/_src/loss.py#L107) without going through it 

Since I removed the smoothing itself, this issue is now fixed in this PR. Should I submit a separate PR to fix this line for the `smooth_labels` as well?